File: teledonJFX/src/main/java/com/example/teledonjfx/repository/VolunteerRepository.java
package com.example.teledonjfx.repository;

import com.example.teledonjfx.model.Volunteer;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class VolunteerRepository implements IVolunteerRepository{
    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();
    public VolunteerRepository(Properties properties){
        logger.info("Initializing VolunteerRepository with properties: {}",properties);
        dbUtils = new JdbcUtils(properties);
    }

    @Override
    public void save(Volunteer elem) {
        logger.traceEntry("saving volunteer {}", elem);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("insert into Volunteers(username,password) values (?,?)")){
            preparedStatement.setString(1,elem.getUsername());
            preparedStatement.setString(2,elem.getPassword());
            int result = preparedStatement.executeUpdate();
            logger.traceExit("Saved {} instances",result);
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        logger.traceExit();

    }

    @Override
    public void delete(Integer id) {
        logger.traceEntry("deleting volunteer {}", id);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("delete from Volunteers where id = ?")){
            preparedStatement.setInt(1,id);
            int result = preparedStatement.executeUpdate();
            logger.traceExit("Deleted {} instances",result);
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        logger.traceExit();
    }

    @Override
    public void update(Volunteer elem, Integer id) {
        logger.traceEntry("updating volunteer {}", id);
        Connection connection = dbUtils.getConnection();
        String sql = String.format("update Volunteers set username = ?, password = ? where id = %d",id);
        try(PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
            preparedStatement.setString(1,elem.getUsername());
            preparedStatement.setString(2,elem.getPassword());
            int result = preparedStatement.executeUpdate();
            logger.trace("Updated {} instances", result);
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        logger.traceExit();
    }

    @Override
    public Volunteer findById(Integer id) {

        logger.traceEntry("finding volunteer with id {}", id);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("select * from Volunteers where id = ?")){
            preparedStatement.setInt(0,id);
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                if(resultSet.next()){
                    int idVolunteer = resultSet.getInt("id");
                    String usernameVolunteer = resultSet.getString("username");
                    String passwordVolunteer = resultSet.getString("password");
                    Volunteer volunteer = new Volunteer(usernameVolunteer,passwordVolunteer);
                    volunteer.setId(idVolunteer);
                    logger.traceExit(volunteer);
                    return volunteer;
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        logger.traceExit("No volunteer found with id {}", id);
        return null;
    }

    @Override
    public Iterable<Volunteer> findAll() {

        logger.traceEntry("finding all volunteers {}");
        Connection connection = dbUtils.getConnection();
        List<Volunteer> volunteers = new ArrayList<>();
        try(PreparedStatement preparedStatement = connection.prepareStatement("select * from Volunteers")) {
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                while(resultSet.next()){
                    int idVolunteer = resultSet.getInt("id");
                    String usernameVolunteer = resultSet.getString("username");
                    String passwordVolunteer = resultSet.getString("password");
                    Volunteer volunteer = new Volunteer(usernameVolunteer,passwordVolunteer);
                    volunteer.setId(idVolunteer);
                    volunteers.add(volunteer);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        logger.traceExit(volunteers);
        return volunteers;
    }

    @Override
    public int size() {


        logger.traceEntry();
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("select count(*) as [SIZE] from Volunteers")){
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                if(resultSet.next()){
                    logger.traceExit(resultSet.getInt("SIZE"));
                    return resultSet.getInt("SIZE");
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        return 0;
    }

    @Override
    public Volunteer getByUsernameAndPassword(String user, String pass) {
        logger.traceEntry("finding volunteer by username and password");
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("select * from Volunteers where username = ? and password = ?")){
            preparedStatement.setString(1, user);
            preparedStatement.setString(2, pass);
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                if(resultSet.next()){
                    int idVolunteer = resultSet.getInt("id");
                    String usernameVolunteer = resultSet.getString("username");
                    String passwordVolunteer = resultSet.getString("password");
                    Volunteer volunteer = new Volunteer(usernameVolunteer,passwordVolunteer);
                    volunteer.setId(idVolunteer);
                    logger.traceExit(volunteer);
                    return volunteer;
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.out.println("Error DB " + e);
        }
        logger.traceExit("No volunteer found");
        return null;
    }
}

File: SocketTeledon/persistence/VolunteerRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;
using log4net;
using teledonCS.model;

namespace teledonCS.repository
{
    public class VolunteerRepository : IVolunteerRepository
    {
        private static readonly ILog log = LogManager.GetLogger("VolunteerRepository");
        IDictionary<String, string> props;

        public VolunteerRepository(IDictionary<string, string> props)
        {
            log.Info("Creating VolunteerRepository");
            this.props = props;
        }

        public void delete(int key)
        {
            log.InfoFormat("Deleting volunteer");
            IDbConnection con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Volunteers where id = @key";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@key";
                paramId.Value = key;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();
                if (dataR == 0)
                {
                    log.Error("No volunteer deleted");
                    throw new Exception("No volunteer deleted!");
                }
                else
                {
                    log.InfoFormat("Volunteer deleted");
                }
            }
            log.InfoFormat("Exiting delete");
        }

        public List<Volunteer> findAll()
        {
            log.InfoFormat("Finding all Volunteers");
            IDbConnection con = DBUtils.getConnection();
            IList<Volunteer> volunteers = new List<Volunteer>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select id,username,password from Volunteers";
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idV = dataR.GetInt32(0);
                        String usernameV = dataR.GetString(1);
                        String passwordV = dataR.GetString(2);
                        Volunteer volunteer = new Volunteer(idV, usernameV, passwordV);
                        volunteers.Add(volunteer);
                    }
                }
            }
            log.InfoFormat("Exiting findAll");

            return (List<Volunteer>) volunteers;
        }

        public Volunteer getByUsernameAndPassword(string username, string password)
        {
            log.InfoFormat("Entering find volunteer by username{0} and password{1}",username,password);
            IDbConnection con = DBUtils.getConnection() ?? throw new ArgumentNullException("DBUtils.getConnection()");

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select id,username,password from Volunteers where username = @user and password = @pass";
                IDbDataParameter paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@user";
                paramUsername.Value = username;
                comm.Parameters.Add(paramUsername);

                IDbDataParameter paramPassword = comm.CreateParameter();
                paramPassword.ParameterName = "@pass";
                paramPassword.Value = password;
                comm.Parameters.Add(paramPassword);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idV = dataR.GetInt32(0);
                        String usernameV = dataR.GetString(1);
                        String passwordV = dataR.GetString(2);
                        Volunteer volunteer = new Volunteer(idV, usernameV, passwordV);
                        log.InfoFormat("Exiting findOne with value{0}", volunteer);
                        return volunteer;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value{0}", null);
            return null;
        }

        public Volunteer findOne(int key)
        {
            log.InfoFormat("Entering findOne with value{0}",key);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select id,username,password from Volunteers where id = @key";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@key";
                paramId.Value = key;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idV = dataR.GetInt32(0);
                        String usernameV = dataR.GetString(1);
                        String passwordV = dataR.GetString(2);
                        Volunteer volunteer = new Volunteer(idV, usernameV, passwordV);
                        log.InfoFormat("Exiting findOne with value{0}", volunteer);
                        return volunteer;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value{0}", null);
            return null;
        }

        public void save(Volunteer el)
        {
            log.InfoFormat("Saving volunteer");
            var con = DBUtils.getConnection();
            using(var comm=con.CreateCommand())
            {
                comm.CommandText = "insert into Volunteers values(@username,@password)";
                var paramUserName = comm.CreateParameter();
                paramUserName.ParameterName = "@username";
                paramUserName.Value = el.username;
                comm.Parameters.Add(paramUserName);
                
                var paramPassword = comm.CreateParameter();
                paramPassword.ParameterName = "@password";
                paramPassword.Value = el.password;
                comm.Parameters.Add(paramPassword);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    log.Error("No volunteer saved");
                    throw new Exception("No volunteer added!!");
                }
                else
                {
                    log.InfoFormat("Volunteer saved");
                }
            }
            log.InfoFormat("Exiting save");
        }

        public void update(Volunteer el)
        {
            log.InfoFormat("Updating volunteer");
            var con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText =
                    "update Volunteers set username = @usernameNew, password = @passwordNew where id = @id";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = el.id;
                comm.Parameters.Add(paramId);

                var paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@usernameNew";
                paramUsername.Value = el.username;
                comm.Parameters.Add(paramUsername);

                var paramPassword = comm.CreateParameter();
                paramPassword.ParameterName = "@passwordNew";
                paramPassword.Value = el.password;
                comm.Parameters.Add(paramPassword);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                {
                    log.Error("No volunteer updated");
                    throw new Exception("No volunteer updated!!");
                }
                else
                {
                    log.InfoFormat("Volunteer updated");
                }
            }

            log.InfoFormat("Exiting update");
        }
    }
}
